src/any_agent/serving/server.py

```diff
@@ -31,33 +33,49 @@ def _get_a2a_app(
 
 
 def _create_server(
-    app: A2AStarletteApplication, host: str, port: int, log_level: str = "warning"
+    app: A2AStarletteApplication,
+    host: str,
+    port: int,
+    endpoint: str,
+    log_level: str = "warning",
 ) -> uvicorn.Server:
-    config = uvicorn.Config(app.build(), host=host, port=port, log_level=log_level)
+    root = endpoint.lstrip("/").rstrip("/")
+    a2a_app = app.build()
+    internal_router = Starlette(routes=[Mount(f"/{root}", routes=a2a_app.routes)])
+
+    config = uvicorn.Config(internal_router, host=host, port=port, log_level=log_level)
     return uvicorn.Server(config)
 
 
 async def serve_a2a_async(
-    server: A2AStarletteApplication, host: str, port: int, log_level: str = "warning"
+    server: A2AStarletteApplication,
+    host: str,
+    port: int,
+    endpoint: str,
+    log_level: str = "warning",
 ) -> tuple[asyncio.Task[Any], uvicorn.Server]:
     """Provide an A2A server to be used in an event loop."""
-    uv_server = _create_server(server, host, port)
+    uv_server = _create_server(server, host, port, endpoint, log_level)
```

Comment:
same as earlier, I wonder if turning this into **kwargs might make life easier

Reply:
(see above)